File: CourseRegistrationSystem/src/Data/TranscriptDto.java
package Data;

import Domain.*;


// A data transfer object for creating transcript json files
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.ArrayList;

@JsonIgnoreProperties(ignoreUnknown = true)
public class TranscriptDto {
    public String studentNo;
    public int completedCredits;
    public ArrayList<GradeDto> grades = new ArrayList<>();


    public TranscriptDto(Student student){
        studentNo = student.getStudentNo();
        Transcript transcript = student.getTranscript();
        completedCredits = transcript.getCompletedCredits();
        for (Grade grade : transcript.getGrades()){
            GradeDto gradeDto = new GradeDto(grade);
            grades.add(gradeDto);
        }
    }

    public TranscriptDto(){};

}

File: Iteration3/Data/TranscriptDto.py
class GradeDto:
    def __init__(self, course_code, num_grade):
        self.courseCode = course_code
        self.numGrade = num_grade


class TranscriptDto:
    def __init__(self, student=None):
        if student:
            self.studentNo = student.get_student_no()
            transcript = student.get_transcript()
            self.completedCredits = transcript.get_completed_credits()
            self.grades = [GradeDto(grade.course_code, grade.num_grade) for grade in transcript.get_grades()]
        else:
            self.studentNo = None
            self.completedCredits = 0
            self.grades = []